Source/RapyutaSimulationPlugins/Private/Tools/SimulationState.cpp

```diff
@@ -83,26 +84,26 @@ void ASimulationState::GetEntityStateSrv(UROS2GenericSrv* Service)
 
     FROSGetEntityState_Response Response;
     Response.success = false;
-    if (Entities.Contains(Request.name))
+    if (SimulationStateData->Entities.Contains(Request.name))

```

Comment:
Worth pointing out that there are _a lot_ of map lookups for this entity and Request.reference_frame. It would be better to pull the actor out into a local variable and do operations on it that way.